src/port1.0/portconfigure.tcl

```diff
@@ -77,7 +77,15 @@ option_proc configure.objcxxflags portconfigure::stdlib_trace
 proc portconfigure::should_add_stdlib {} {
     set has_stdlib [expr {[option configure.cxx_stdlib] ne ""}]
     set is_clang [string match *clang* [option configure.cxx]]
-    return [expr {$has_stdlib && $is_clang}]
+    # GCC also supports -stdlib starting with GCC 12 (and devel)
+    set is_gcc_with_stdlib no
+    if { [string match macports-gcc-* [option configure.compiler]] } {
+        set gcc_ver [string map {"macports-gcc-" ""} [option configure.compiler]]
```

Comment:
`string range` might be a better fit than `string map` here?

Reply:
done

Reply:
In the end went with a combination of `split` and `lindex`